### DISCUSSION TITLE
NixOS Flake Home-Manager: there is no $XDG_RUNTIME_DIR/hypr directory

### DISCUSSION THREAD

**Discussion by Ziqi-Yang** [hyprwm/Hyprland#6019] [Q&A] [answered]
Recently I switched my configuration to use flake based hyprland configuration (before this I uesd hyprland on NixOS with home-manager, without Flake). However, after the switching, I cannot run `hyprctl version` and it says: 
```bash
❯ hyprctl version
Couldn't connect to /run/user/1000/hypr/v0.39.1_1715424281/.socket.sock. (3)
```

Additional information:
```bash
❯ echo $XDG_RUNTIME_DIR/hypr
/run/user/1000/hypr
❯ ls /run/user/1000/hypr
ls: cannot access '/run/user/1000/hypr': No such file or directory
❯ ls /tmp/hypr
v0.39.1_1714896252       v0.39.1_1714960575       v0.39.1_1715069631.lock  v0.39.1_1715220007.lock  v0.39.1_1715242042       v0.39.1_1715403057.lock  v0.39.1_1715424281 ...

❯ waybar # works well, but it use `/tmp/hypr`
...
[2024-05-11 18:50:39.526] [info] Hyprland IPC starting
[2024-05-11 18:50:39.526] [warning] $XDG_RUNTIME_DIR/hypr does not exist, falling back to /tmp/hypr
...

❯ readlink -f (which hyprctl)
/nix/store/bvljn1ahjldsg1i975zhqaxq2zax87n5-hyprland-0.40.0+date=2024-05-10_ed3a888/bin/hyprctl

❯ readlink -f (which Hyprland)
/nix/store/bvljn1ahjldsg1i975zhqaxq2zax87n5-hyprland-0.40.0+date=2024-05-10_ed3a888/bin/Hyprland


```

--- 
My configuration  
**Before**: 
in `nixos` module:
```nix
programs.hyprland.enable = true;
```
in `home-manager` module:
```nix
  wayland.windowManager.hyprland.enable = true;
  # and other configurations 
```
**After**:
in `nixos` module: (https://git.sr.ht/~meow_king/nixos-config/tree/0b99a556468159a23ac7406129bd6790346f29ef/item/hosts/meowos/default.nix#L144-147)
```nix
  programs = {
    hyprland = {
      enable = true;
      package = inputs.hyprland.packages.${pkgs.system}.hyprland;
    };
  };
```
in `flake` outupt (https://git.sr.ht/~meow_king/nixos-config/tree/0b99a556468159a23ac7406129bd6790346f29ef/item/hosts/default.nix#L27-54)
```nix
  nixosConfigurations = {
    meowos = lib.nixosSystem {
      system = "x86_64-linux";
      specialArgs = {inherit inputs outputs gVars mylib secrets;};
      modules = [
        ./meowos

        home-manager.nixosModules.home-manager {
          home-manager = {
            useGlobalPkgs = true;
            useUserPackages = true;
            users.${gVars.username} = import ../home/linux/desktop-develop.nix;
            extraSpecialArgs = { inherit inputs gVars mylib secrets; };
            sharedModules = [
              hyprland.homeManagerModules.default
            ];
          };
        }
      ];
    };
  };
```
the configuration inside `home-manager` module doesn't change: (https://git.sr.ht/~meow_king/nixos-config/tree/0b99a556468159a23ac7406129bd6790346f29ef/item/home/linux/desktop/hyprland.nix#L37)

---

Thank you very much for your help!

**Comment by fufexan** [comment, marked as answer]
0.39.1 is the version in Nixpkgs. The flake has 0.40.0, which is the only version (currently) that includes the $XDG_RUNTIME_DIR change. Are you positive your hyprland input is up to date? Try running a `nix flake update`.

**Reply by Ziqi-Yang** [reply]
Sorry for the late reply. Just now I just rechecked my configuration, and I find in my configuration, my login manager actually uses `pkgs.hyprland` as its input, which causes the problem. Thank you very much!
